int32 id
int32 vid
int32 plid
int32 type # Don't use wide character
int32 linkid
